plat/common/arm/time.c

```diff
@@ -51,6 +51,13 @@ static const char * const arch_timer_list[] = {
 	NULL
 };
 
+#if defined(CONFIG_PLAT_XEN)
```

Comment:
I suggest a bit of the re-phrasing in the commit message:
```
The arch_timer implemented for arm32 is not suitable for aarch64 because
it doesn't use interrupts, so it makes system to wait forever during sleep
call.
Add support for generic interrupt based timer.
```